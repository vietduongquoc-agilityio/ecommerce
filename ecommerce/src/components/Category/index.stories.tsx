import React from "react";
import Category from "../Category";
import { Meta, StoryObj } from "@storybook/react";

const meta: Meta<typeof Category> = {
  title: "Components/Category",
  component: Category,
  tags: ["autodocs"],
  parameters: {
    layout: "centered",
  },
};

export default meta;

type Story = StoryObj<typeof Category>;

export const Default: Story = {
  render: () => <Category />,
};
