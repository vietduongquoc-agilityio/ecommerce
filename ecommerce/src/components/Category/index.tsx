"use client";

import { Box, Text } from "@radix-ui/themes";
import Image from "next/image";
import dining from "@/assets/Images/Dining-room.png";
import living from "@/assets/Images/Living-room.png";
import bed from "@/assets/Images/Bed-room.png";
import { colors, fonts } from "@/themes";

const Category = () => {
  return (
    <Box width="100%">
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          flexDirection: "column",
          alignItems: "center",
        }}
      >
        <Text
          style={{
            fontSize: fonts.size.xl,
            fontWeight: fonts.weight.bold,
            color: colors.secondary.black,
          }}
        >
          Browse The Range
        </Text>
        <Text
          style={{
            fontSize: fonts.size.md,
            fontWeight: fonts.weight.regular,
            color: colors.secondary.splight,
            marginTop: "10px",
          }}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Text>
      </div>
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          gap: "40px",
          marginTop: "50px",
        }}
      >
        {[
          { src: dining, label: "Dining" },
          { src: living, label: "Living" },
          { src: bed, label: "Bedroom" },
        ].map((item) => (
          <Box
            key={item.label}
            style={{
              display: "flex",
              flexDirection: "column",
              alignItems: "center",
              gap: "30px",
              fontSize: fonts.size.lg,
              fontWeight: fonts.weight.semiBold,
              cursor: "pointer",
              transition: "transform 0.3s ease",
            }}
            onMouseOver={(e) =>
              (e.currentTarget.style.transform = "scale(1.1)")
            }
            onMouseOut={(e) => (e.currentTarget.style.transform = "scale(1)")}
          >
            <Image src={item.src} alt={`${item.label}-icon`} />
            <p>{item.label}</p>
          </Box>
        ))}
      </div>
    </Box>
  );
};

export default Category;
