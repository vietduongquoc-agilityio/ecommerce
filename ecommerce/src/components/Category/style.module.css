.title {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 30px;
  font-size: 24px;
  font-weight: 600;
  cursor: pointer;
  transition: transform 0.3s ease;
}

.title:hover {
  transform: scale(1.1);
}

.heading {
  font-weight: 700;
  font-size: 32px;
}

.desc {
  font-weight: 400;
  font-size: 20px;
  color: #666666;
}

