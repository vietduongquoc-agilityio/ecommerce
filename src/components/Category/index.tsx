"use client";

import { Box, Text } from "@radix-ui/themes";
import Image from "next/image";

// Images
import dining from "@/assets/Images/Dining.jpg";
import living from "@/assets/Images/Living.jpg";
import bed from "@/assets/Images/Bed-room.jpg";
import { colors, fonts } from "@/themes";

const Category = () => {
  return (
    <Box width="100%">
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          flexDirection: "column",
          alignItems: "center",
          marginBottom: "50px",
        }}
      >
        <Text
          style={{
            fontSize: fonts.size.xl,
            fontWeight: fonts.weight.bold,
            color: colors.secondary.black,
          }}
        >
          Browse The Range
        </Text>
        <Text
          style={{
            fontSize: fonts.size.md,
            fontWeight: fonts.weight.regular,
            color: colors.secondary.splight,
            marginTop: "10px",
          }}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Text>
      </div>
      <div
        style={{
          display: "flex",
          width: "1183px",
          justifyContent: "space-between",  
          margin: "0 auto",
        }}
      >
        {[
          { src: dining, label: "Dining" },
          { src: living, label: "Living" },
          { src: bed, label: "Bedroom" },
        ].map((item) => (
          <Box
            key={item.label}
            style={{
              display: "flex",
              flexDirection: "column",
              alignItems: "center",
              fontSize: fonts.size.lg,
              fontWeight: fonts.weight.semiBold,
              cursor: "pointer",
              transition: "transform 0.3s ease",
            }}
            onMouseOver={(e) =>
              (e.currentTarget.style.transform = "scale(1.1)")
            }
            onMouseOut={(e) => (e.currentTarget.style.transform = "scale(1)")}
          >
            <Image
              src={item.src}
              alt={`${item.label}-icon`}
              style={{
                width: "375px",
                height: "480px",
                borderRadius: "10px",
                alignItems: "center",
                marginBottom: "25px"
              }}
            />
            <p>{item.label}</p>
          </Box>
        ))}
      </div>
    </Box>
  );
};

export default Category;
