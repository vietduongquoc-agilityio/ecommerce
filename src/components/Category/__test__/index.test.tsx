import { render } from "@testing-library/react";
import Category from "../index";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();
describe("Category Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();

    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });
  });
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Category />);
    expect(asFragment()).toMatchSnapshot();
  });
});
