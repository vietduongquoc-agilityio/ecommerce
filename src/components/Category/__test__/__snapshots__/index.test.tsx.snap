// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rt-Box rt-r-w"
    style="--width: 100%;"
  >
    <div
      style="display: flex; justify-content: center; flex-direction: column; align-items: center; margin-bottom: 50px;"
    >
      <span
        class="rt-Text"
        style="font-size: 34px; font-weight: 700; color: rgb(58, 58, 58);"
      >
        Browse The Range
      </span>
      <span
        class="rt-Text"
        style="font-size: 20px; font-weight: 400; color: rgb(159, 159, 159); margin-top: 10px;"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </span>
    </div>
    <div
      style="display: flex; width: 1183px; justify-content: space-between; margin: 0px auto;"
    >
      <div
        class="rt-Box"
        style="display: flex; flex-direction: column; align-items: center; font-size: 24px; font-weight: 600; cursor: pointer; transition: transform 0.3s ease;"
      >
        <img
          alt="Dining-icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent; width: 375px; height: 480px; border-radius: 10px; align-items: center; margin-bottom: 25px;"
          width="40"
        />
        <p>
          Dining
        </p>
      </div>
      <div
        class="rt-Box"
        style="display: flex; flex-direction: column; align-items: center; font-size: 24px; font-weight: 600; cursor: pointer; transition: transform 0.3s ease;"
      >
        <img
          alt="Living-icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent; width: 375px; height: 480px; border-radius: 10px; align-items: center; margin-bottom: 25px;"
          width="40"
        />
        <p>
          Living
        </p>
      </div>
      <div
        class="rt-Box"
        style="display: flex; flex-direction: column; align-items: center; font-size: 24px; font-weight: 600; cursor: pointer; transition: transform 0.3s ease;"
      >
        <img
          alt="Bedroom-icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent; width: 375px; height: 480px; border-radius: 10px; align-items: center; margin-bottom: 25px;"
          width="40"
        />
        <p>
          Bedroom
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
